src/annotator/components/toolbar.js

```diff
@@ -0,0 +1,135 @@
+import classnames from 'classnames';
+import propTypes from 'prop-types';
+import { createElement } from 'preact';
+
+function ToolbarButton({
+  buttonRef,
+  extraClasses,
+  label,
+  icon,
+  onClick,
+  selected,
+}) {
+  const handleClick = event => {
```

Comment:
Can you elaborate on why we need to snarf up events here?

Reply:
Yes. The sidebar has document-level event handlers that close the sidebar when clicks happen in the document. In some cases third-party code in the page may also handle clicks in some way. See also https://github.com/hypothesis/client/pull/2106.